lib/awful/rules.lua

```diff
@@ -207,23 +211,169 @@ function rules.matches_list(c, _rules)
     return false
 end
 
---- Apply awful.rules.rules to a client.
--- @client c The client.
-function rules.apply(c)
 
-    local props = {}
+-- Contains the sources.
+-- The elements are ordered "first in, first executed". Thus, the higher the
+-- index, the higher the priority. Each entry is a table with a `name` and a
+-- `callback` field. This table is exposed for debugging purpose. The API
+-- is private and should be modified using the public accessors.
```

Comment:
1) This comment talks about index-based priorities. I thought the idea was to get rid of that?
2) If this is private, then why is it public (i.e.: not `local`) *and* documented?

Reply:
> This comment talks about index-based priorities.

Because it's the *result* of the sort, so it's index based ;)

> then why is it public (i.e.: not local) 

done

> and documented?

It is not publicly documented, it's `--` instead of `---` ;). The `awful.rules` code is already a spagatti mess just to be able to do everything in the right order, at least it's a well documented (and necessary...) mess.